ssl/quic/quic_port.c

```diff
@@ -45,30 +44,47 @@ static void port_rx_pre(QUIC_PORT *port);
  * @var validation_token::timestamp
  * Time that the validation token was minted.
  *
+ * @var validation_token::odcid
+ * An original connection ID (`QUIC_CONN_ID`) used to identify the QUIC
+ * connection. This ID helps associate the token with a specific connection.
+ * This will only be valid for validation tokens from RETRY packets.
+ *
+ * @var validation_token::rscid
+ * DCID that the client will use as the DCID of the subsequent initial packet
+ * i.e the "new" DCID.
+ * This will only be valid for validation tokens from RETRY packets.
+ *
  * @var validation_token::remote_addr_len
  * Length of the following character array.
  *
  * @var validation_token::remote_addr
  * A character array holding the raw address of the client requesting the
  * connection.
- *
- * @var validation_token::odcid
- * An original connection ID (`QUIC_CONN_ID`) used to identify the QUIC
- * connection. This ID helps associate the token with a specific connection.
- *
- * @var validation_token::rscid
- * The original retry source connection ID (`QUIC_CONN_ID`).
  */
 typedef struct validation_token {
-    int is_retry;
+    unsigned char is_retry;
     OSSL_TIME timestamp;
-    size_t remote_addr_len;
-    unsigned char *remote_addr;
-    /* Only set for retry tokens */
     QUIC_CONN_ID odcid;
     QUIC_CONN_ID rscid;
+    size_t remote_addr_len;
+    unsigned char *remote_addr;
 } QUIC_VALIDATION_TOKEN;
 
+/*
+ * The QUIC_CONN_ID has unused bytes in it (the struct is maximally sized to a
+ * 20 byte id array, but we only use an 8 byte one). We can save 12 bytes per
+ * conn_id by copying only conn_id.id_len + 1 bytes (for the id_len field) and
+ * restoring it in the same way on validation.
+*/
+#define QUIC_CONN_ID_LEN 9
```

Comment:
Not sure why you need this, the marshalled data should be 1 byte (sizeof(cid.id_len)), plus cid.id_len bytes.  We shouldn't statically define the size. If you compute it using the size of the id_len field + the number of bytes it tells you the id is, then we're future proofed against larger cid sizes